include/pybind11/cast.h

```diff
@@ -889,14 +889,25 @@ template <typename type> class type_caster<std::reference_wrapper<type>> {
     operator std::reference_wrapper<type>() { return subcaster.operator subcaster_cast_op_type&(); }
 };
 
+/// Call class-specific delete if it exists or global otherwise. Can also be an overload set.
+template <typename T, typename = void_t<decltype(static_cast<void (*)(void *)>(T::operator delete))>>
+void call_operator_delete(T *p) { T::operator delete(p); }
+
+inline void call_operator_delete(void *p) { ::operator delete(p); }
+
 #define PYBIND11_TYPE_CASTER(type, py_name) \
     protected: \
         type value; \
     public: \
         static PYBIND11_DESCR name() { return type_descr(py_name); } \
-        static handle cast(const type *src, return_value_policy policy, handle parent) { \
+        template <typename _T, enable_if_t<std::is_same<type, remove_cv_t<_T>>::value, int> = 0> \
+        static handle cast(_T *src, return_value_policy policy, handle parent) { \
```

Comment:
Good point; I was copying the `typename _T` for the `cast_op_type` just below; I'll change both to `T_`.